Sources/ApolloWebSocket/WebSocketTransport.swift

```diff
@@ -297,7 +303,14 @@ extension WebSocketTransport: WebSocketDelegate {
       // re-send the subscriptions whenever we are re-connected
       // for the first connect, any subscriptions are already in queue
       for (_,msg) in self.subscriptions {
-        write(msg)
+        if allowSendingDuplicates {
+          write(msg)
+        } else {
+          // search duplicate message from the queue
+          // do not send multiple times
+          let id = queue.first { $0.value == msg }?.key
+          write(msg, id: id)
```

Comment:
I'm confused - this looks like it'll just `write` the same `message` with the same `id` rather than bailing if a duplicate is found, but the comment says "do not send multiple times". How is this preventing duplicates from being sent?

Reply:
I agree. I removed misleading comment.

The actual place where it checks for previous message is from line 225.

Basically how it should work if we not allowing to send duplicates is:
1. Try to find previous message (currently just by content because we don't match subscription IDs in the queue)
	1. a If found then use the ID from the queue
	1. b If not found then send nil for `ID` to `write`
3. In `write` function
	1. If `ID` is not `nil` we update previous message
	2. If `ID` is null then add new message to the queue, this did happen before
	3. If queue is empty then add first entry

Ideally we need to somehow link together `subscriptions` dictionary and internal `queue` of messages. But I would like to do it gradually. What do you think?

Reply:
OK so right now the `write` queue is doing all the work. Got it. Thanks for changing the comment to make that clearer.